python/simulation.py

```diff
@@ -639,6 +638,20 @@ def __init__(self,
         self.fragment_stats = None
         self._output_stats = os.environ.get('MEEP_STATS', None)
 
+        if kz_2d == "complex":
+            self.special_kz = True
+            self.force_complex_fields = True
+        elif kz_2d == "real/imag":
+            self.special_kz = True
+            self.force_complex_fields = False
+        elif kz_2d == "3d":
+            self.special_kz = False
+        else:
+            raise ValueError("Invalid kz_2d option: {} not in [complex, real/imag, 3d]".format(kz_2d))
+
+        if self.special_kz and self.k_point is False:
+            self.k_point = Vector3()
+
```

Comment:
Lines 641-653 should be replaced with:

```
        self.special_kz = False
        if self.cell_size.z == 0 and self.k_point.z != 0:
            if kz_2d == "complex":
                self.special_kz = True
                self.force_complex_fields = True
            elif kz_2d == "real/imag":
                self.special_kz = True
                self.force_complex_fields = False
            elif kz_2d == "3d":
                self.special_kz = False
            else:
                raise ValueError("Invalid kz_2d option: {} not in [complex, real/imag, 3d]".format(kz_2d))
```

Setting `kpoint=0` when `special_kz=True` is incorrect since its non-zero z component is what is used in the calculations.